server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -198,6 +203,51 @@ protected final DocsStats docsStats(IndexReader indexReader) {
         return new DocsStats(numDocs, numDeletedDocs, sizeInBytes);
     }
 
+    /**
+     * Returns the {@link DenseVectorStats} for this engine
+     */
+    public DenseVectorStats denseVectorStats() {
+        try (Searcher searcher = acquireSearcher(DOC_STATS_SOURCE, SearcherScope.INTERNAL)) {
+            return denseVectorStats(searcher.getIndexReader());
+        }
+    }
+
+    protected final DenseVectorStats denseVectorStats(IndexReader indexReader) {
+        long valueCount = 0;
+        // we don't wait for a pending refreshes here since it's a stats call instead we mark it as accessed only which will cause
+        // the next scheduled refresh to go through and refresh the stats as well
+        for (LeafReaderContext readerContext : indexReader.leaves()) {
+            // we go on the segment level here to get accurate numbers
+            final SegmentReader segmentReader = Lucene.segmentReader(readerContext.reader());
+            SegmentCommitInfo info = segmentReader.getSegmentInfo();
+            try {
+                valueCount += getDenseVectorValueCount(readerContext.reader());
```

Comment:
Should `segmentReader` be passed there (rather than `readerContext.reader()`) ?

Reply:
We can use the `readerContext.reader()` directly. I pushed https://github.com/elastic/elasticsearch/pull/98275/commits/9437ac064ce6ca3406149111981f9c5befb40d9f